File: microservices/auto_processing_datasets/src/lib/database/base/BaseAdapter.py
from abc import ABC, abstractmethod


class BaseAdapter(ABC):
    """Base adapter interface for database operations."""
    
    @abstractmethod
    def connect(self) -> None:
        """Connect to the database."""
        pass
    
    @abstractmethod
    def disconnect(self) -> None:
        """Disconnect from the database."""
        pass
    
    @abstractmethod
    def find_one(self, collection: str, filter: dict) -> dict | None:
        """Find one document in a collection."""
        pass
    
    @abstractmethod
    def update_one(self, collection: str, filter: dict, update: dict) -> None:
        """Update one document in a collection."""
        pass

File: backend/src/lib/database/base/BaseAdapter.ts
export interface DatabaseAdapter {
  connect(): Promise<void>;
  disconnect(): Promise<void>;
  createCollection(name: string): Promise<void>;
  collectionExists(name: string): Promise<boolean>;
  insertOne(collection: string, document: any): Promise<any>;
  insertMany(collection: string, documents: any[]): Promise<any[]>;
  findOne(collection: string, filter: any): Promise<any | null>;
  findMany(collection: string, filter?: any, options?: any): Promise<any[]>;
  updateOne(collection: string, filter: any, update: any): Promise<any>;
  updateMany(collection: string, filter: any, update: any): Promise<number>;
  deleteOne(collection: string, filter: any): Promise<boolean>;
  deleteMany(collection: string, filter: any): Promise<number>;
  count(collection: string, filter?: any): Promise<number>;
}

